src/numbers/xdr_int.js

```diff
@@ -0,0 +1,251 @@
+    const lo64 = BigInt.asUintN(64, v); // grab btm 64, encode sign
+
+    return xdr.ScVal.scvI128(
+      new xdr.Int128Parts({
+        hi: new xdr.Int64(hi64),
+        lo: new xdr.Uint64(lo64)
+      })
+    );
+  }
+
+  /**
+   * @returns {xdr.ScVal} the integer encoded with `ScValType = U128`
+   * @throws {RangeError} if the value cannot fit in 128 bits
+   */
+  toU128() {
+    this._sizeCheck(128);
+    const v = this.int.toBigInt();
+
+    return xdr.ScVal.scvU128(
+      new xdr.UInt128Parts({
+        hi: new xdr.Uint64(BigInt.asUintN(64, v >> 64n)),
+        lo: new xdr.Uint64(BigInt.asUintN(64, v))
+      })
+    );
+  }
+
+  /**
+   * @returns {xdr.ScVal} the integer encoded with `ScValType = I256`
+   */
+  toI256() {
+    const v = this.int.toBigInt();
+    const hiHi64 = BigInt.asIntN(64, v >> 192n); // keep sign bit
+    const hiLo64 = BigInt.asUintN(64, v >> 128n);
+    const loHi64 = BigInt.asUintN(64, v >> 64n);
+    const loLo64 = BigInt.asUintN(64, v);
+
+    return xdr.ScVal.scvI256(
+      new xdr.Int256Parts({
+        hiHi: new xdr.Int64(hiHi64),
+        hiLo: new xdr.Uint64(hiLo64),
+        loHi: new xdr.Uint64(loHi64),
+        loLo: new xdr.Uint64(loLo64)
+      })
+    );
+  }
+
+  /**
+   * @returns {xdr.ScVal} the integer encoded with `ScValType = U256`
+   */
+  toU256() {
+    const v = this.int.toBigInt();
+    const hiHi64 = BigInt.asUintN(64, v >> 192n); // encode sign bit
+    const hiLo64 = BigInt.asUintN(64, v >> 128n);
+    const loHi64 = BigInt.asUintN(64, v >> 64n);
+    const loLo64 = BigInt.asUintN(64, v);
+
+    return xdr.ScVal.scvU256(
+      new xdr.UInt256Parts({
+        hiHi: new xdr.Uint64(hiHi64),
+        hiLo: new xdr.Uint64(hiLo64),
+        loHi: new xdr.Uint64(loHi64),
+        loLo: new xdr.Uint64(loLo64)
+      })
+    );
+  }
+
+  valueOf() {
+    return this.int.valueOf();
+  }
+
+  toString() {
+    return this.int.toString();
+  }
+
+  toJSON() {
+    return {
+      value: this.toBigInt().toString(),
+      type: this.type
+    };
+  }
+
+  _sizeCheck(bits) {
+    if (this.int.size > bits) {
+      throw RangeError(`value too large for ${bits} bits (${this.type})`);
+    }
+  }
+}
+
+/**
+ * Transforms an opaque {@link xdr.ScVal} into a native bigint, if possible.
+ *
+ * If you then want to use this in the abstractions provided by this module,
+ * you can pass it to the constructor of {@link XdrInt}.
+ *
+ * @example
+ * ```js
+ * let scv = contract.call("add", x, y); // assume it returns an xdr.ScVal
+ * let bigi = scValToInt(scv);
+ *
+ * new ScInt(bigi);           // if you don't care about types, and
+ * new XdrInt('i128', bigi);  // if you do
+ * ```
+ *
+ * @param {xdr.ScVal} scv - the raw XDR value to parse into an integer
+ * @returns {bigint} the native value of this input value
+ *
+ * @throws {TypeError} if the `scv` input value doesn't represent an integer
+ */
+export function scValToInt(scv) {
```

Comment:
should this be refactored to `XdrInt.fromScVal()`, to be symmetric with the `XdrInt.toXYZ()` methods that return ScVal. Noticed usage is referenced in ScInt docs - https://github.com/stellar/js-stellar-base/pull/620/files#diff-b8cf857162926efd3588586c0e321ff3a60ef479b2ab7e9482d7e3c9a3ea82b5R22 and in the index.d.ts

Reply:
Hah, this is the exact discussion from above, here: https://github.com/stellar/js-stellar-base/pull/620#discussion_r1223413318. I decided to move it out because it returns a `bigint` not an `XdrInt`. The thought was that it should be separate because 32-bit support is wonky and users are likelier to want the native value. Wdyt?

Reply:
ah, ok, see that aspect of 64 or larger now, similar to the suggested rename for `scValToBigInt` function, a few adjustments for consideration:

* `XdrLargeInt` as the class name? I think it may help sdk users quickly get the intent, otherwise, 'int' types with no other qualifier in the type name usually hold anything up to a limit. Or add some docs at the class level to drive the point of only 64 bit and up.

* relocate `scValToBigInt` as it's not related to the class, like up to numbers/index.js or something?

Reply:
good suggestions, done :+1: